### ISSUE TITLE
_doing_it_wrong notices on every request: BlockExtension instantiated before init triggers is_page() and textdomain JIT-load via Vipps integration

### ISSUE THREAD

**Issue by Dekadinious** [krokedil/klarna-checkout-for-woocommerce#744]
# `_doing_it_wrong` notices on every request: `BlockExtension` instantiated before `init` triggers `is_page()` and textdomain JIT-load via Vipps integration

## Summary

In 2.20.0 (and several earlier releases), `BlockExtension` is constructed during the `plugins_loaded` action via `KCO::init_gateways()`. Its constructor unconditionally calls `wc_get_page_id( 'checkout' )` via `BlocksUtility::is_checkout_block_enabled()`, which fires the `woocommerce_get_page_id` filter while WP is still pre-`init`.

When the official Vipps/MobilePay for WooCommerce plugin (`woo-vipps`) is active, its `VippsCheckout::maybe_override_checkout_page_id` callback is registered on that filter and:

- calls `is_page()`, and
- lazy-instantiates `WC_Gateway_Vipps`, whose constructor calls `__('…', 'woo-vipps')` and `__('…', 'woocommerce')` via `init_form_fields()`.

Both are pre-`init`, so WP 6.7+ emits `_doing_it_wrong` notices on every request:

- `Function is_page was called incorrectly. Conditional query tags do not work before the query is run.`
- `Translation loading for the woo-vipps domain was triggered too early.`
- `Translation loading for the woocommerce domain was triggered too early.`

## Environment

- WordPress 6.7+
- WooCommerce (current)
- `klarna-checkout-for-woocommerce` 2.20.0 (also reproduced on prior 2.18.x / 2.19.x line)
- `woo-vipps` (Vipps/MobilePay for WooCommerce, current)

## Reproduction

1. Have both KCO ≥ 2.18.x and `woo-vipps` active.
2. Enable WP debug logging.
3. Hit any frontend page or `admin-ajax.php`.
4. Observe three `_doing_it_wrong` notices per request, with `did_action('init') === false` at the point of emission.

## Backtrace

One of several — all share the same shape:

```
is_page() @ wp-content/plugins/woo-vipps/payment/VippsCheckout.class.php:1318
VippsCheckout->maybe_override_checkout_page_id() @ wp-includes/class-wp-hook.php:341
WP_Hook->apply_filters() @ wp-includes/plugin.php:205
apply_filters( 'woocommerce_get_page_id', ... ) @ wp-content/plugins/woocommerce/includes/wc-page-functions.php:84
wc_get_page_id() @ wp-content/plugins/klarna-checkout-for-woocommerce/src/Utility/BlocksUtility.php:20
Krokedil\KustomCheckout\Utility\BlocksUtility::is_checkout_block_enabled() @ wp-content/plugins/klarna-checkout-for-woocommerce/blocks/src/BlockExtension.php:50
Krokedil\KustomCheckout\Blocks\BlockExtension->init_checkout_block() @ wp-content/plugins/klarna-checkout-for-woocommerce/blocks/src/BlockExtension.php:41
Krokedil\KustomCheckout\Blocks\BlockExtension->__construct() @ wp-content/plugins/klarna-checkout-for-woocommerce/klarna-checkout-for-woocommerce.php:304
KCO->init_gateways() @ wp-content/plugins/klarna-checkout-for-woocommerce/klarna-checkout-for-woocommerce.php:167
KCO->init()
WP_Hook->apply_filters() (plugins_loaded)
```

## Root cause (KCO side)

`klarna-checkout-for-woocommerce.php`:

```php
protected function __construct() {
    add_action( 'plugins_loaded', array( $this, 'init' ) );  // hooks pre-init
    ...
}

public function init() {
    $this->init_gateways();
}

public function init_gateways() {
    ...
    $this->block_extension = new BlockExtension();   // pre-init
}
```

`blocks/src/BlockExtension.php`:

```php
public function __construct() {
    $this->init_checkout_block();   // unconditional pre-init query lookup
}

public function init_checkout_block() {
    if ( ! BlocksUtility::is_checkout_block_enabled() ) {   // calls wc_get_page_id()
        return;
    }
    ...
}
```

`is_checkout_block_enabled()` calling `wc_get_page_id()` from a `plugins_loaded` callback is the trigger. Even without Vipps in the mix, doing query work and option-driven lookups during `plugins_loaded` is at minimum surprising; with Vipps's `woocommerce_get_page_id` filter callback present, it produces the visible WP 6.7 notices.

## Suggested fix

Defer `BlockExtension` instantiation (or at least the `is_checkout_block_enabled()` check) until `init` or, more naturally, `woocommerce_blocks_loaded` — which is what the rest of `init_checkout_block()` already hooks into. Something like:

```php
public function init_gateways() {
    ...
    if ( $autoloader_result ) {
        add_action( 'init', function () {
            $this->block_extension = new BlockExtension();
        }, 5 );
    }
}
```

…or move the gating check inside the `woocommerce_blocks_loaded` callbacks rather than into the constructor. Either resolves the pre-`init` `wc_get_page_id` call and removes the cascade through Vipps.

**Comment by mntzrr** [comment]
Thank you for taking your time to report this issue! I created a PR (#745) addressing it. It should be included in the next release.
